src/RazorSdk/Targets/Sdk.Razor.CurrentVersion.targets

```diff
@@ -50,9 +50,11 @@ Copyright (c) .NET Foundation. All rights reserved.
         <_TargetingNET50OrLater>true</_TargetingNET50OrLater>
         <_TargetingNET60OrLater>true</_TargetingNET60OrLater>
         <UseRazorSourceGenerator Condition="'$(UseRazorSourceGenerator)' == '' ">true</UseRazorSourceGenerator>
+        <!-- The incremental source generator API is disabled on VS 16.11 and below so we disable it there. -->
+        <UseRazorSourceGenerator Condition="'$(VisualStudioVersion)' &lt; '17.0'">false</UseRazorSourceGenerator>
```

Comment:
This feels very iffy. What happens if we don't do this and have the user figure this out? We're already claiming 6.0 requires VS 2022. (Also I don't think the comparison works - it's comparing two strings).

Reply:
I based this off the examples in https://dotnet.grep.app/search?q=%27%24%28VisualStudioVersion%29%27%20%26lt%3B&filter[lang][0]=XML. I believe it works because it does a lexographic comparison.

> What happens if we don't do this and have the user figure this out? 

Users will end up getting weird errors about types not being available when they build in VS. Since the errors aren't meant for end-users their pretty hostile (e.g. "Cannot load type IIncrementalGenerator from assembly").

I think it's localized enough that it's worth doing as a quality of life improvement.

Reply:
Could we leave this out for now? We can always add it back if we get a lot of feedback. The problem is that turning off the source generator also has a noticeable runtime behavior difference. It would be very spooky to see if this just because an older version of VS opened this.